docs/Information-Flow.md

```diff
@@ -0,0 +1,88 @@
+# Information Flow
+
+It's recommended you understand the [basics of `sbp`](Style-Guide.md#sbp) before moving on because everything that will come next, works around it.
+
+The simplest way to explain this is by starting to explain a couple of functions you'll see together a lot of times across the project. These two functions are the core to make any logical change at Group Income:
+
+```js
+// Example: Update a group minimum income to 150
+
+// 1. Message - Create a set of data to be later processed
+const message = await sbp(
+  'gi.contracts/group/updateSettings/create',
+  { incomeProvided: 150 },
+  currentGroupId
+)
+
+// 2. Mutation - Send the message to the server and update the store (Vuex) accordingly
+await sbp('backend/publishLogEntry', message)
+```
+
+Now, let me explain in more detail what really happens in each function:
+
+##### 1. Create a Message
+
+The SBP selector [_domain_ `gi.contracts/`](../frontend/model/contracts/group.js) is based on a [common Contract](../frontend/model/Contract.js). Each action (ex: `group/updateSettings`) expects 2 arguments: the `data` and `contractId`.
+- `data`: an object passed to the action when is processed.
+- `contractId`: the current group hash where the action should take effect on.
+
+This group action creates and returns a [GIMessage](../shared/GIMessage.js) based on the information passed (`data` and `contractId`).
+
+> **Note:** A _Contract_ is a set of information with a specific structure where the client can subscribe to. It can be a group, a user profile (identity), the mailbox or any other thing. All current contracts can be found at [`model/contacts/`](../frontend/model/contracts/group.js).
+
+##### 2. Perform a Mutation
+
+With this new message, we need to send it to the server...
+
+```js
+await sbp('backend/publishLogEntry', message)
+```
+
+###### 2.1 Communication with the server (database)
+
+Long story short, the sbp selector `'backend/publishLogEntry'` sends a POST request to the server with the _message_ serialized. The server, eventually, through a _websocket_, returns the _message_ back to the client, deserializes and dispatches it to Vuex.
+
+Then, [the dispatch event is handled by Vuex at model/state.js](../frontend/model/state.js#L302). There, the event is verified and, if all goes right, the message is processed through `sbp`, with the given `selector` (the action), the state based on the `contractId` and the `data`.
```

Comment:
This is a bit too much detail, I would give a more general description of the idea of what happens, rather than precisely what happens. I.e., something along the lines of:

> GIMessages ("events") are used to represent either the creation of a contract, or an action performed upon that contract. When a client sends these messages to the server, they are sent back to everyone who is subscribed to that contract. Then, upon receiving a message/event, each client uses it to update their local copy of the contract state.